### PR TITLE
Upgrade To React 18 with Suspense SSR

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**

### CODE REVIEW

File: nextjs-ssr/checkout/package.json [hunk 1]
```diff
@@ -11,10 +11,10 @@
     "nextjs-shared": "*",
     "@module-federation/nextjs-ssr": "0.1.0-rc.7",
     "lodash": "4.17.21",
-    "next": "12.1.0",
+    "next": "12.1.7-canary.16",
```
Comment (ScriptedAlchemy) [inline nextjs-ssr/checkout/package.json:14]: Bug in next with reactRoot, only fixed in canary

File: nextjs-ssr/checkout/pages/index.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
 import dynamic from 'next/dynamic';
 const page = import('home/home');
-
-const Page = dynamic(() => import('home/home'));
+import { Suspense } from 'react';
+const Component = dynamic(() => import('home/home'), { suspense: true });
+const Page = props => (
+  <Suspense>
```
Comment (ScriptedAlchemy) [inline nextjs-ssr/checkout/pages/index.js:6]: pages need suspense boundaries if dynamic is suspense based

File: nextjs-ssr/home/components/nav.js [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
 import Link from 'next/link';
 import dynamic from 'next/dynamic';
 
-export const HelloWorld = dynamic(() => import('./helloWorld').then(mod => mod.HelloWorld), {
-  ssr: true,
-});
+export const HelloWorld = dynamic(
+  async () => {
+    const { HelloWorld } = await import('./helloWorld');
+    return { default: HelloWorld };
```
Comment (ScriptedAlchemy) [inline nextjs-ssr/home/components/nav.js:8]: react lazy only accepts .default

**Comment by integrate-your-mind** [comment]
wen merge?

**Comment by ScriptedAlchemy** [comment]
Next doesnt fully support Suspense SSR yet, so there is FOUT

**Pull request closed without merging**